src/main/java/de/tum/in/www1/artemis/web/rest/tutorialgroups/TutorialGroupResource.java

```diff
@@ -0,0 +1,279 @@
+
+        Optional<User> studentToDeregister = userRepository.findOneWithGroupsAndAuthoritiesByLogin(studentLogin);
+        if (studentToDeregister.isEmpty()) {
+            throw new EntityNotFoundException("User", studentLogin);
+        }
+        tutorialGroupService.deregisterStudent(studentToDeregister.get(), tutorialGroupFromDatabase);
+        return ResponseEntity.noContent().build();
+    }
+
+    /**
+     * POST tutorial-groups/:tutorialGroupId/register/:studentLogin : register a student to a tutorial group.
+     *
+     * @param tutorialGroupId the id of the tutorial group
+     * @param studentLogin    the login of the student to register
+     * @return the ResponseEntity with status 204 (NO_CONTENT)
+     */
+    @PostMapping("/tutorial-groups/{tutorialGroupId}/register/{studentLogin:" + Constants.LOGIN_REGEX + "}")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<Void> registerStudent(@PathVariable Long tutorialGroupId, @PathVariable String studentLogin) {
+        log.debug("REST request to register {} student to tutorial group : {}", studentLogin, tutorialGroupId);
+        var tutorialGroupFromDatabase = this.tutorialGroupRepository.findByIdElseThrow(tutorialGroupId);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, tutorialGroupFromDatabase.getCourse(), null);
+
+        Optional<User> userToRegister = userRepository.findOneWithGroupsAndAuthoritiesByLogin(studentLogin);
+        if (userToRegister.isEmpty()) {
+            throw new EntityNotFoundException("User", studentLogin);
+        }
+        var student = userToRegister.get();
+        if (!student.getGroups().contains(tutorialGroupFromDatabase.getCourse().getStudentGroupName())) {
+            throw new BadRequestAlertException("The user is not a student of the course", ENTITY_NAME, "userNotPartOfCourse");
+        }
+
+        tutorialGroupService.registerStudent(userToRegister.get(), tutorialGroupFromDatabase);
+        return ResponseEntity.noContent().build();
+    }
+
+    /**
+     * POST tutorial-groups/:tutorialGroupId/register-multiple/" : Register multiple users to the tutorial group
+     *
+     * @param tutorialGroupId the id of the tutorial group to which the users should be registered to
+     * @param studentDtos     the list of students who should be registered to the tutorial group
+     * @return the list of students who could not be registered for the tutorial group, because they could NOT be found in the Artemis database as students of the tutorial group course
+     */
+    @PostMapping("/tutorial-groups/{tutorialGroupId}/register-multiple")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<Set<StudentDTO>> registerMultipleStudentsToTutorialGroup(@PathVariable Long tutorialGroupId, @RequestBody Set<StudentDTO> studentDtos) {
```

Comment:
Do you even need the singular endpoints (line 195 and 218)? I could imagine you could use this endpoint to register one student as well. That would mean less code to maintain.